src/main/java/net/coderbot/iris/Iris.java

```diff
@@ -91,6 +100,21 @@ private void loadExternalShaderpack(String name) {
 		logger.info("Using shaderpack: " + name);
 	}
 
+	private static Optional<Path> loadExternalZipShaderpack(Path shaderpackPath) {
+		try {
+			FileSystem zipSystem = FileSystems.newFileSystem(shaderpackPath, Iris.class.getClassLoader());
+			Path root = zipSystem.getRootDirectories().iterator().next();//should only be one root directory for a zip shaderpack
+			return Files.walk(root)
+				.filter(Files::isDirectory)
+				.filter(path -> path.endsWith("shaders"))
+				.findFirst();
```

Comment:
Why not just use [getPath](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/FileSystem.html#getPath(java.lang.String,java.lang.String...)) here?

Reply:
It's because while testing I found some zipped packs to have a folder before the `shaders` dir, so for example, `Trippy-Shaderpack-master.zip` could have a directory of `Trippy-Shaderpack-master.zip/Trippy-Shaderpack-master/shaders` so it's not always clear if the `shaders` dir is directly inside the zip or nested one folder

Reply:
Couldn’t this cause issues with chopapic, where it has multiple shaders in one zip?

Reply:
Ah, right. Maybe it could be better to look for `shaders` dir, and if it doesn't exist, fall back to this method? Not sure though.

In any case, it would definitely help to have a comment explaining that in the code.

Reply:
Okay I think i've added everything, can you take a final look around?